internal/services/serviceconnector/service_connector_app_service_resource.go

```diff
@@ -72,7 +72,6 @@ func (r AppServiceConnectorResource) Arguments() map[string]*schema.Schema {
 		"vnet_solution": {
 			Type:     pluginsdk.TypeString,
 			Optional: true,
-			Default:  string(servicelinker.VNetSolutionTypePrivateLink),
```

Comment:
this is a breaking change, it will need to be removed in 4.0, could we wrap this change in the 4.0 flag

Reply:
Hi @katbyte , I think this can be considered as a bug instead of breaking change. Per discussion with service team, there's some misunderstanding during the development of this resource at first time. The default value of this attribute should be null instead of a `private link`, passing `private link` as default will cause resource creating failure. Since it's not described in documentation, so it's not disclosed to our customers.

Reply:
@jiaweitao001 except if someone is relying on the behaviour right now this will break it for them. vs it not working till someone removes the default. can we please wrap this change in the 4.0 flag

Reply:
@katbyte , wrapped with 4.0 flag.

Reply:
Hi @katbyte , synced with service team of this resource, they told me that there's no customer impact of this resource yet because this default behavior bug is actually blocking users from leveraging TF to create resources. Since there are no users and this is this root cause, I think maybe we'd better fix it right away. Thanks!

Reply:
Sure @jiaweitao001 - if they can confirm that they that works for me